### PR TITLE
19 hydra

### LINKED ISSUE #19: use Hydra to load configurations and manage hyperparameters for experiments (M11)

### PR DESCRIPTION
**Pull request by bruijnis-b**
This pull request introduces Hydra-based configuration management, replacing hardcoded and CLI argument-based configuration with structured YAML files. The changes centralize all key hyperparameters, model, data, and training options in the `configs/hydra` directory, and refactor the codebase to load these settings via Hydra. This improves reproducibility, maintainability, and flexibility for experiments. The CLI entry points for training and data preparation now use Hydra, and the model and data modules are simplified to rely on configuration objects.

**Key changes:**

**Hydra Configuration Integration:**
- Added a full suite of Hydra YAML configuration files under `configs/hydra`, covering data, model, optimizer, trainer, and paths. These files define all key parameters (batch size, learning rate, feature/target selection, etc.) in a central, version-controlled location. (`configs/hydra/config.yaml`, `configs/hydra/data/default.yaml`, `configs/hydra/model/tabular_classifier.yaml`, `configs/hydra/optimizer/adam.yaml`, `configs/hydra/optimizer/sgd.yaml`, `configs/hydra/paths/default.yaml`, `configs/hydra/trainer/default.yaml`)

- Updated `pyproject.toml` to require `hydra-core` as a dependency.

**Refactoring Data and Model Code for Hydra:**
- Refactored `src/diabetic_classification/data.py`:
    - Removed CLI argument parsing and replaced with Hydra configuration loading.
    - Simplified the `DiabetesHealthDataset` class by removing support for excluded features and requiring explicit feature/target attribute lists from the config. 
    - The CLI entry point for data preparation now uses Hydra.

- Refactored `src/diabetic_classification/train.py`:
    - Replaced Typer-based CLI with Hydra-based configuration.
    - Training parameters (data paths, batch size, optimizer, etc.) are now read from the config files.
    - The CLI entry point for training now uses Hydra.

**Model Configuration and Optimizer Logic:**
- Updated `DiabetesClassifier` in `model.py` to accept model and optimizer configuration objects, and to construct the optimizer based on the config (supporting both Adam and SGD). 

These changes make the codebase more modular and experiment-friendly, allowing for easy adjustment of hyperparameters and settings without code changes.

### CODE REVIEW

File: src/diabetic_classification/model.py [hunk 1]
```diff
@@ -57,31 +57,26 @@ class DiabetesClassifier(LightningModule):
...
-            weight_decay: float = 1e-4,
             output_dim: int = 1,
-            hidden_dims: tuple[int, int] = (128, 64),
-            dropout: float = 0.2,
     ) -> None:
         super().__init__()
-        self.lr = lr
-        self.weight_decay = weight_decay
+        self.optimizer_cfg = optimizer_cfg
+        print(self.optimizer_cfg)
```
Comment (Mathiaz21) [inline src/diabetic_classification/model.py]: Maybe use logging here instead of a print, something like 
`from loguru import logger`
...
`logger.info(f"Optimizer config: {self.optimizer_cfg}"`

**Review by Mathiaz21 — APPROVED** [review]
LGTM !
